src/cgutils.cpp

```diff
@@ -1253,11 +1253,77 @@ static Value *data_pointer(jl_codectx_t &ctx, const jl_cgval_t &x, Type *astype
             data = boxed(ctx, x);
         }
     }
-    if (data->getType() != astype)
+    if (astype && data->getType() != astype)
         data = emit_bitcast(ctx, data, astype);
     return decay_derived(data);
 }
 
+static void emit_memcpy_llvm(jl_codectx_t &ctx, Value *dst, Value *src,
+                             uint64_t sz, unsigned align, bool is_volatile, MDNode *tbaa)
+{
+    // If the types are small and simple, use load and store directly.
+    // Going through memcpy can cause LLVM (e.g. SROA) to create bitcasts between float and int
+    // that interferes with other optimizations.
+    if (sz <= 64) {
```

Comment:
1 cache line at most? This is just the max vector size we need to deal with atm.